src/include/sof/ipc.h

```diff
@@ -213,6 +218,11 @@ struct ipc_data {
 
 	/* PM */
 	uint32_t pm_target_state;	/* SOF_PM_STATE_D.. */
+
+	/* register based communication */
+	uint32_t ipc_payload;		/* SOF_IPC_PAYLOAD_... */
+	uint32_t reg1;
+	uint32_t reg2;
```

Comment:
I don't get why this is needed. I thought we would either
A) get the info directly from the memory windows
B) uncompress the information from the IPC register

In both cases, we'd handle the same structure in the end, so not sure why we would keep this information in a intermediate structure.